langroid/agent/tool_message.py

```diff
@@ -22,6 +23,55 @@
 )
 from langroid.utils.types import is_instance_of
 
+K = TypeVar("K")
+
+
+def remove_if_exists(k: K, d: dict[K, Any]) -> None:
+    """Removes key `k` from `d` if present."""
+    if k in d:
+        d.pop(k)
+
+
+def format_schema_for_strict(schema: Any) -> None:
```

Comment:
This is used on all schemas for strict mode (both output formats and strict tool schemas), and should ensure compatibility (however, there may be edge cases as I'm not aware of precise documentation of the OpenAI JSON schema subset for strict mode). 

This may not be equivalent to the original schema, but will be in typical cases (e.g. "allOf" will occasionally be generated by Pydantic but generally only with a singleton set of types, in which case it is equivalent to "anyOf"), and the loss of "default" means that any defaults can only be expressed in an external prompt.

Given these requirements, is it safe to enable strict mode by default when supported?

Reply:
I think it's fine -- 

- we could automatically retry with non-strict mode if there's an error (I think you already have this?)

We can also have a config `use_strict_mode_if_available` to control this behavior, and if there's an error we can print a friendly warning saying strict mode was used since it was allowed, and suggest turning off this flag.